ui/components/FeedActions.vue

```diff
@@ -0,0 +1,179 @@
+<template>
+  <ui-menu
+    class="feed-actions"
+    align="right"
+  >
+    <div slot="trigger">
+      <slot />
+    </div>
+    <div
+      slot="content"
+      class="actions-menu"
+    >
+      <div
+        v-for="button in buttons"
+        :key="button.action"
+        class="action-button__wrapper"
+      >
+        <div
+          v-if="button.separated"
+          class="horizontal-divider"
+        />
+        <ui-button
+          v-if="!button.hide"
+          :ref="button.action"
+          class="action-button"
+          :background-color="COLORS_DICT.TRANSPARENT"
+          :color="button.color"
+          :disabled="button.disabled"
+          height="32px"
+          width="156px"
+          @clicked="button.handler"
+        >
+          <div class="button-content">
+            <ui-icon
+              :icon-name="button.loading ? 'connectLoaderAnimated' : button.icon"
+              :color="button.color"
+              size="18"
+            />
+            <span>{{ button.text }}</span>
+          </div>
+        </ui-button>
+      </div>
+    </div>
+  </ui-menu>
+</template>
+
+<script setup>
+import { useToolkit } from '@cloudblueconnect/connect-ui-toolkit/tools/vue/toolkitPlugin';
+import { computed, reactive } from 'vue';
+
+import { useRequest } from '~/composables/api';
+import { COLORS_DICT } from '~/constants/colors';
+import { STATUSES_DICT } from '~/constants/statuses';
+
+const props = defineProps({
+  feed: {
+    type: Object,
+    required: true,
+  },
+});
+
+const emit = defineEmits(['enabled', 'disabled', 'uploaded', 'deleted']);
+
+const toolkit = useToolkit();
+
+const enableAction = useRequest(toolkit, true);
+const disableAction = useRequest(toolkit, true);
+const deleteAction = useRequest(toolkit, true);
+const forceUploadAction = useRequest(toolkit, true);
+
+const feedId = computed(() => props.feed.id);
+const isFeedDisabled = computed(() => props.feed.status === STATUSES_DICT.DISABLED);
+
+const buttons = reactive([
+  {
+    action: 'disable',
+    color: COLORS_DICT.TEXT,
+    text: 'Disable',
+    icon: 'googleToggleOffBaseline',
+    loading: disableAction.loading,
+    hide: isFeedDisabled,
+    handler: async () => {
```

Comment:
wdyt about moving this into a different method?

Reply:
Good point, I extracted all handlers outside the `buttons` variable